### PR TITLE
EASY-2385: wrapper for dataset.xml and files.xml

### PR DESCRIPTION
**Pull request by vesaakerman**
Fixes EASY-2385

#### When applied it will
* wrap `dataset.xml` and `files.xml` inside `bagmetadata` xml-file

#### By submitting this pull request I confirm that
* [x] all files contain licenses (`mvn license:format`)
* [x] the project compiles (`mvn clean install`)
* [x] all unit tests pass (`mvn test`)
* [x] the changes have been tested on `deasy`

#### Related pull requests on github
repo                       | PR
-------------------------- | -----------------
easy-                      | [PR#](PRlink)

### CODE REVIEW

File: src/main/scala/nl/knaw/dans/easy/transform/XmlWrapper.scala [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+package nl.knaw.dans.easy.transform
+
+import scala.xml.Node
+
+object XmlWrapper {
+
+  private val xml = <bagmetadata/>
+
+  def wrap(datasetXml: Node, filesXml: Node): Node = {
+    xml.copy(child = xml.child ++ datasetXml ++ filesXml)
```
Comment (rvanheest) [inline src/main/scala/nl/knaw/dans/easy/transform/XmlWrapper.scala]:
```suggestion
    <bagmetadata>
      { datasetXml }
      { filesXml }
    </bagmetadata>
```

Comment (rvanheest) [reply src/main/scala/nl/knaw/dans/easy/transform/XmlWrapper.scala]: and then you also don't need the `xml` value above anymore...

Comment (rvanheest) [reply src/main/scala/nl/knaw/dans/easy/transform/XmlWrapper.scala]: perhaps we should reference the schema here as well? @lindareijnhoudt can you take a look at that?

File: src/test/scala/nl/knaw/dans/easy/transform/XmlWrapperSpec.scala [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+  "wrap" should "return dataset.xml and files.xml in a wrapping xml" in {
+    val datasetXml = XML.loadFile(dataset_open)
+    val filesXml = XML.loadFile(files_open)
+    val result = XmlWrapper.wrap(datasetXml, filesXml)
+    result.child should have size 2
+    result \ "DDM" should have size 1
+    result \ "files" should have size 1
+    result.child.head.label shouldBe "DDM"
+    result.child(1).label shouldBe "files"
```
Comment (rvanheest) [inline src/test/scala/nl/knaw/dans/easy/transform/XmlWrapperSpec.scala]: How about

```scala
result.child.map(_.label) should contain inOrderOnly ("DDM", "files")
```

**Review by rvanheest — COMMENTED** [review]
a first review...

File: src/main/scala/nl/knaw/dans/easy/transform/XmlWrapper.scala (hunk 1 above)
Comment (lindareijnhoudt) [reply src/main/scala/nl/knaw/dans/easy/transform/XmlWrapper.scala]: i agree this is more clear. and it also allows to declare the namespace

\<bagmetadata xmlns="http://easy.dans.knaw.nl/schemas/bag/metadata/bagmetadata/">

**Comment by rvanheest** [comment]
It isn't shown in this PR, but the build fails... Please fix!

File: src/test/scala/nl/knaw/dans/easy/transform/XmlWrapperSpec.scala [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    val filesXml = XML.loadFile(files_open)
+    val result = XmlWrapper.wrap(datasetXml, filesXml)
+
+    validate(result, File(bagmetadataSchema))
+  }
+
+  private def validate(xmlFile: Node, xsdFile: File): Unit = {
+    val schemaFactory = SchemaFactory.newInstance(XMLConstants.W3C_XML_SCHEMA_NS_URI)
+    val schema: Schema = schemaFactory.newSchema(xsdFile.toJava)
+    val validator: Validator = schema.newValidator()
```
Comment (jo-pol) [inline src/test/scala/nl/knaw/dans/easy/transform/XmlWrapperSpec.scala:62]: Have a look at [assume](https://github.com/DANS-KNAW/easy-deposit-api/blob/50a26ef09ce21faf54656fc0d771265450babf5a/src/test/scala/nl.knaw.dans.easy.deposit/docs/FilesXmlSpec.scala#L59) and [loadSchema](https://github.com/DANS-KNAW/easy-deposit-api/blob/50a26ef09ce21faf54656fc0d771265450babf5a/src/test/scala/nl.knaw.dans.easy.deposit/docs/package.scala#L54-L61)

**Pull request merged**